src/components/datatable/useTableData.js

```diff
@@ -0,0 +1,262 @@
+    } else if (Array.isArray(aggregationType) && aggregationType.length) {
+        customFields = aggregationType.map((type) => {
+            let roundFn = null
+            if (data?.length) {
+                const precision = getPrecision(data.map((d) => d[type]))
+                roundFn = numberPrecision(precision)
+            }
+            return {
+                name: toTitleCase(`${type} ${title}`),
+                dataKey: type,
+                roundFn,
+                type: TYPE_NUMBER,
+            }
+        })
+    }
+
+    return [INDEX, NAME, ID, TYPE]
+        .map((field) => defaultFieldsMap()[field])
+        .concat(customFields)
+}
+
+const EMPTY_AGGREGATIONS = {}
+const EMPTY_LAYER = {}
+
+export const useTableData = ({ layer, sortField, sortDirection }) => {
+    const allAggregations = useSelector((state) => state.aggregations)
+    const aggregations = allAggregations[layer.id] || EMPTY_AGGREGATIONS
+
+    const {
+        layer: layerType,
+        aggregationType,
+        legend,
+        styleDataItem,
+        data,
+        dataFilters,
+        headers: layerHeaders,
+        serverCluster,
+    } = layer || EMPTY_LAYER
+
+    const dataWithAggregations = useMemo(
+        () =>
+            data
+                .map((d, i) => ({
+                    index: i,
+                    ...d,
+                }))
+                .filter((d) => !d.properties.hasAdditionalGeometry)
+                .map((d, i) => ({
+                    ...(d.properties || d),
+                    ...aggregations[d.id],
+                    index: d.index,
+                    i,
+                })),
+        [data, aggregations]
+    )
+
+    const headers = useMemo(() => {
+        if (!layerType || !dataWithAggregations.length) {
+            return []
+        }
+
+        switch (layerType) {
+            case THEMATIC_LAYER:
+                return getThematicHeaders()
+            case EVENT_LAYER:
+                return getEventHeaders({ layerHeaders, styleDataItem })
+            case ORG_UNIT_LAYER:
+                return getOrgUnitHeaders()
+            case EARTH_ENGINE_LAYER:
+                return getEarthEngineHeaders({
+                    aggregationType,
+                    legend,
+                    data: dataWithAggregations,
+                })
+            case FACILITY_LAYER:
+                return getFacilityHeaders()
+            default:
+                // TODO - throw error?
+                return []
+        }
+    }, [
+        layerType,
+        aggregationType,
+        legend,
+        styleDataItem,
+        dataWithAggregations,
+        layerHeaders,
+    ])
+
+    const rows = useMemo(() => {
+        if (!dataWithAggregations.length || !headers?.length) {
+            return []
+        }
+
+        const filteredData = filterData(dataWithAggregations, dataFilters)
+
+        //sort
+        filteredData.sort((a, b) => {
+            a = a[sortField]
+            b = b[sortField]
```

Comment:
I believe our normal ESLint rules prevent variable reassignment. Could we work with some local variables here too? Sth like `sortFieldA` and `sortFieldB`?